framework/include/markers/BoundaryPreservedMarker.h

```diff
@@ -0,0 +1,34 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ErrorFractionMarker.h"
+
+/*
+  Do not coarsen any element touching a given boundary in order to preserve the boundary
+  libMesh will delete all boundaries for the elements to be coarsened.
+*/
+class BoundaryPreservedMarker : public ErrorFractionMarker
```

Comment:
@dewenyushu I think a better design would be to make this a separate marker like `ComboMarker`. That would allow you to combine it with arbitrary markers.

Reply:
Sounds like a great idea. Let me look into this a bit more:)